pymatgen/entries/entry.py

```diff
@@ -0,0 +1,101 @@
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+"""
+This module implements the base Entry class, which is the basic entity
+that can be used to store calculated information. Other Entry classes 
+such as ComputedEntry and PDEntry inherit from this class.
+"""
+from pymatgen.core.composition import Composition
+from monty.json import MSONable
+from abc import ABCMeta, abstractmethod
+
+
+__author__ = "Shyue Ping Ong, Anubhav Jain, Ayush Gupta"
+__copyright__ = "Copyright 2020, The Materials Project"
+__version__ = "1.1"
+__maintainer__ = "Shyue Ping Ong"
+__email__ = "shyuep@gmail.com"
+__status__ = "Production"
+__date__ = "Mar 02, 2020"
+
+
+class Entry(MSONable, metaclass=ABCMeta):
+    """
+    A lightweight Entry object containing various types 
+    of data associated with a specific chemical composition.
+
+    """
+
+    def __init__(self,
+                 composition: Composition,
+                 energy: float):
+        """
+        Initializes an Entry.
+
+        Args:
+            composition (Composition): Composition of the entry. For
+                flexibility, this can take the form of all the typical input
+                taken by a Composition, including a {symbol: amt} dict,
+                a string formula, and others.
+            energy (float): Energy of the entry.
+            correction (float): A correction to be applied to the energy.
+                This is used to modify the energy for certain analyses.
+                Defaults to 0.0.
+        """
+        self._energy = energy
+        self.composition = Composition(composition)
+
+    @property
+    def is_element(self) -> bool:
+        """
+        :return: Whether composition of entry is an element.
+        """
+        return self.composition.is_element
+
+    @property
+    @abstractmethod
+    def energy(self) -> float:
+        """
+        :return: the energy of the entry. Each subclass must define its own energy property.
```

Comment:
@ayushsgupta Given the changes to the main docstring, we can now delete 'Each subclass must define its own energy property.'